modules/programs/terminator.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.terminator;
+
+  toValue = val:
+    if val == null then
+      "None"
+    else if val == true then
+      "True"
+    else if val == false then
+      "False"
+    else
+      ''"${toString val}"'';
+
+  toConfigObject = let
+    toKey = depth: key:
+      if depth == 0 then key else toKey (depth - 1) "[${key}]";
+    go = depth: obj:
+      flatten (mapAttrsToList (key: val:
+        if isAttrs val then
+          [ (toKey depth key) ] ++ go (depth + 1) val
+        else
+          [ "${key} = ${toValue val}" ]) obj);
+  in obj: concatStringsSep "\n" (go 1 obj);
+
+in {
+  meta.maintainers = [ maintainers.chisui ];
+
+  options.programs.terminator = {
+    enable = mkEnableOption "terminator tiling terminal emulator";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.terminator;
+      defaultText = literalExample "pkgs.terminator";
+      description = "terminator package to install.";
+    };
+
+    config = mkOption {
+      default = null;
+      description = "configuration for terminator";
+      type = types.nullOr (types.submodule {
```

Comment:
```suggestion
      default = { };
      description = "Configuration for terminator";
      type = types.submodule {
```

Reply:
I used the `null` case to toggle whether or not home-manager should manage the configuration. isn't this the way to do it?

Reply:
I would just do the check based off of whether the config is `{ }` or not.